nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -867,15 +871,18 @@ protected void remove(String rootId) {
 
         // find all sub-docs
         Set<String> removedIds = new HashSet<>();
-        Set<String> retentionActiveIds = new HashSet<>();
+        Set<String> undeletableIds = new HashSet<>();
         Set<String> targetIds = new HashSet<>();
         Map<String, Object[]> targetProxies = new HashMap<>();
 
         Consumer<State> collector = state -> {
             String id = (String) state.get(KEY_ID);
             removedIds.add(id);
+            if (TRUE.equals(state.get(KEY_IS_UNDELETABLE))) {
+                undeletableIds.add(id);
+            }
             if (TRUE.equals(state.get(KEY_IS_RETENTION_ACTIVE))) {
```

Comment:
Two if can be merged.

Reply:
Diff is clearer without it, also in the future the KEY_IS_RETENTION_ACTIVE case will be removed.